scripts/gogenerate.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+unset -v progdir
+case "${0}" in
+*/*) progdir="${0%/*}";;
+*) progdir=.;;
+esac
+git grep -l '^//go:generate ' | sort -t/ | (
+	gogenerate() {
+		case $# in 0) return;; esac
+		go generate "$@"
+	}
+	unset -v prevdir dir file
+	prevdir=
+	set --
+	while read -r file
```

Comment:
please add some comments to elaborate the block.

Reply:
Done (in the utility script).